src/robotFactory.js

```diff
@@ -1,10 +1,81 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x,
+      y,
+    };
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name},`
+    + ` Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, { x = 0, y = 0, z = 0 }, chipVersion) {
+    super(name, weight, {
+      x,
+      y,
+    }, chipVersion);
+
+    this.coords.z = z;
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, { x = 0, y = 0, z = 0 },
+    chipVersion, maxLoadWeight, currentLoad = null) {
+    super(name, weight, {
+      x,
+      y,
+      z,
+    }, chipVersion);
```

Comment:
Same about repeated logic. 
If you move one property to the new line - the other properties should take one line each

```suggestion
  constructor(
  name, 
  weight, 
  coords,
  chipVersion, 
  maxLoadWeight, 
  currentLoad = null) {
    super(name, weight, coords, chipVersion);
```